hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -501,6 +507,115 @@ public void setPathProperties(final Path path,
+   * of {@link BlobProperty}.
+   *
+   * @param blobPath blobPath for which property information is requried
+   * @param tracingContext object of TracingContext required for tracing server calls.
+   * @return BlobProperty for the given path
+   * @throws AzureBlobFileSystemException exception thrown from
+   * {@link AbfsClient#getBlobProperty(Path, TracingContext)} call
+   */
+  private BlobProperty getBlobProperty(Path blobPath,
+      TracingContext tracingContext) throws AzureBlobFileSystemException {
+    AbfsRestOperation op = client.getBlobProperty(blobPath, tracingContext);
+    BlobProperty blobProperty = new BlobProperty();
+    final AbfsHttpOperation opResult = op.getResult();
+    blobProperty.setIsDirectory(opResult
+        .getResponseHeader(X_MS_META_HDI_ISFOLDER) != null);
+    blobProperty.setUrl(op.getUrl().toString());
+    blobProperty.setCopyId(opResult.getResponseHeader(X_MS_COPY_ID));
+    blobProperty.setPath(blobPath);
+    blobProperty.setCopySourceUrl(opResult.getResponseHeader(X_MS_COPY_SOURCE));
+    blobProperty.setStatusDescription(
+        opResult.getResponseHeader(X_MS_COPY_STATUS_DESCRIPTION));
+    blobProperty.setCopyStatus(getCopyStatus(opResult));
+    blobProperty.setContentLength(
+        Long.parseLong(opResult.getResponseHeader(CONTENT_LENGTH)));
+    return blobProperty;
+  }
+
+  @org.apache.hadoop.classification.VisibleForTesting
+  String getCopyStatus(final AbfsHttpOperation opResult) {
+    return opResult.getResponseHeader(X_MS_COPY_STATUS);
+  }
+
+  /**
+   * Call server API for ListBlob on the blob endpoint. This API returns a limited
+   * number of blobs and provide a field called as NextMarker which is reference to
+   * next list of blobs for the query. Server expects that the client calls this API
+   * in loop with the NextMarker received in previous iteration of backend call for
+   * the same request.
+   *
+   * @param sourceDirBlobPath path from where the list of blob is required.
+   * @param prefix Optional value to be provided. If provided, API call would have
+   * prefix = given value. If not provided, the API call would have prefix =
+   * sourceDirBlobPath.
+   * @param tracingContext object of {@link TracingContext}
+   * @param maxPerServerCallResult define how many blobs can client handle in server response.
+   * In case maxResult <= 5000, server sends number of blobs equal to the value. In
+   * case maxResult > 5000, server sends maximum 5000 blobs.
+   * @param maxResult defines maximum blobs the method should process
+   * @param absoluteDirSearch defines if (true) it is blobList search on a
+   * definitive directory, if (false) it is blobList search on a prefix.
+   *
+   * @return List of blobProperties
+   *
+   * @throws AbfsRestOperationException exception from server-calls / xml-parsing
+   */
+  public List<BlobProperty> getListBlobs(Path sourceDirBlobPath,
+      String prefix, TracingContext tracingContext, Integer maxPerServerCallResult,
```

Comment:
What is the need for maxPerServerCallResult ? Today driver has a config control on page size, can the same not be the default to use.

Reply:
define how many blobs can client handle in server response. In case maxResult <= 5000, server sends number of blobs equal to the value. In case maxResult > 5000, server sends maximum 5000 blobs.
It gives the caller of the method to define, how many blobs should server send back. For example, to check if a path is a directory or not. caller will give maxPerServerCallResult = 2.

Reply:
Only additional and internal requirement in v2 when it come to list call is to do a small listing, like 1 or 2, otherwise it needs to fallback to the original max where listing goes for the current max which is already a config. Hence having a separate maxServerCallResult should be avoided. Such minor controls for same knobs present in config will create layered and complex handling in later with respect to precedence etc. 
getListBlobs() only needs to take listPageSize. if this is lesser than configured, pass this, else use the config for list page already present.

Reply:
Agreed. removed maxPerServerCallResult